### DISCUSSION TITLE
Export from MoBi to PKsim / Save parameters updates on MoBi

### DISCUSSION THREAD

**Discussion by SamLeDruide** [Open-Systems-Pharmacology/Forum#1527] [How-Tos] [answered]
Hi,

I want to change the value of some physiological parameters (plasma volumes / volumes of interstitial compartments of some organs) of a PBPK model. To access this level of granularity, I need to work on MoBi. I understood how to export simulations from PKSIM to MoBi, to modify the parameters on MoBi, and then run the simulations on MoBi.

What I don't understand is how to save the changes I make in MoBi; I don't see a way to update an individual's parameters and then save those changes, without having to start over if I leave MoBi. Also, I don't see how to export this new individual from MoBi to PKSIM, nor how to export the newly created simulations. 

Could you please tell me if this is possible? Thank you in advance for your answer,

Best,

Samuel.

**Comment by msevestre** [comment]
@SamLeDruide Make sure to ask question in the forum. I Moved the issue for you

As to your question, you can change parameters in PKSim directly. Only `structural changes` really require MoBi.  

What parameter are you trying to change in PKSim? 

Once you are in MoBi , you lose the concept of PKSim building block. The only item you can export back to pksim is a simulation (save as pkml, and import as pkml in PKSim)

**Reply by SamLeDruide** [reply]
Thanks @msevestre, and sorry for writing in the wrong place.

I want to do two things in the model : 

1/ For some organs, I want to increase the volume contained in the interstitial compartment and reduce the volume of the plasma compartment accordingly.

2/ I want to segment the "muscle" compartment, to be able to characterize some muscles in particular, so I add a "muscle X" and "other muscles" compartment.

I thought MoBi was the only way to do that, am I wrong ?

Thanks a lot.

**Reply by msevestre** [reply]
If you are changing the structure, then MoBI is the way to go. Your initial message mentioned changing "values". 
You'll have to stay in MoBi (or use the simulation in PKSim as a blackbox but usage is very much limited to running population simulations)

**Reply by SamLeDruide** [reply]
Alright, thanks a lot. Do you confirm that I can perform the 1/ step on PKSIM directly?

**Reply by StephanSchaller** [reply]
Yes, you can do Step 1 in PK.Sim directly

**Reply by SamLeDruide** [reply]
Ok, thank you. I thought of doing it on PKSIM by updating the interstitial fraction value, but it is not possible because the box is grayed out, unlike on MoBi where I can do it directly. Also, I don't see where to change the plasma compartment in the organ on PKsim. Is there something I am missing ?

**Reply by msevestre** [reply]
@SamLeDruide Hum you are absolutely right. Those parameters are marked as readonly in PKSim. Probably sthg that should be removed 
So for now, the easiest for you is probably to go to MoBi (or use R)

**Reply by SamLeDruide** [reply]
Alright. Thanks for your very fast answers, I wish you a great day.

Samuel.

**Reply by SamLeDruide** [reply]
Alright. Thanks for your very fast answers, I wish you a great day.

Samuel.

**Reply by SamLeDruide** [reply]
Hi again,

I am very sorry to bother you.

Can you confirm that it is normal that : 

When I send to MoBi a simulation from PKSIM (I tried with clopidogrel & paracetamol models available on github) : 
1/ no simulation window appear in MoBi, and I have to do Modeling > Simulation > and configure with the spatial strucure, molecules etc. that was imported when I sent the simulation to MoBi.
2/ Right clicking on Organism, or any compartment of the organism is not opening any window.
3/ Right clicking on the Simulation only provide me three options : "close", "Close All Documents", "Close All But This".

Because of this, I cannot save the modifications that I do in MoBi. When I want to save my work, I do File > Save as > and I tried extension mbp3 and also .pkml. When I load a mbp3, it never conserved the modifications of parameters I did, and also, it's not opening any simulation window. When I open a pkml modified in MoBi doing File > Open a Simulation, same result. And when I open a project save as pkml in PKSIM, it return me an error message "Data at the root level is invalid. Line 1, position 1".

I am a bit lost there because I saw in other discussions some screenshots of right clicks on simulation window in MoBi, with a possibility of export in PKSIM. Also, I tried 3 days ago the same process with paracetamol model, and I was able to add compartment in MoBi by right clicking on Organism, or in subcompartments. Also, when I create a new spatial structure in MoBi, their I can add new compartments. 

I unistalled and reinstalled all package, and I did run the Installation validator who returns me no error. 

Do you have any clue of what's happening?

**Reply by msevestre** [reply]
Can you post some screenshots?
And also maybe the pkml file that you are using? Loading a pkml file is business as usual and whatever you do typically in MoBi is available

**Reply by SamLeDruide** [reply]
For instance I try with this simulation : (from the model of clopidogrel available on git)
When on PKSIM I do right click on the simulation and send it to MoBi, MoBi appears like this : 
<img width="782" alt="image" src="https://github.com/Open-Systems-Pharmacology/Forum/assets/133014111/1021ddf8-440e-4a14-824a-8decd18e057b">
Then on MoBi I do Modeling > Simulation > giving a name and OK > Run. It gives me this : 
<img width="957" alt="image" src="https://github.com/Open-Systems-Pharmacology/Forum/assets/133014111/7d8902c1-0f91-4fb2-984b-a563fd860547">
And I right clicking on "organism" did not opened any window : 
<img width="956" alt="image" src="https://github.com/Open-Systems-Pharmacology/Forum/assets/133014111/77e94e82-b680-4cab-a526-70787b114c04">


For the PKML file : it's not supported by github how can I send it to you ?

**Reply by msevestre** [reply]
You can zip it

**Reply by SamLeDruide** [reply]
[14.3.A_D_E_Savu et al 2016_male.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/11456517/14.3.A_D_E_Savu.et.al.2016_male.zip)
Oh yes! thanks a lot.

**Reply by msevestre** [reply]
ok so you are missing the actual building block view
go to the VIEWS RIbbon 
![image](https://github.com/Open-Systems-Pharmacology/Forum/assets/1041237/03b1652c-604f-4578-9acb-6c4d9e551ee3)

and click on the Building block icon and then simulation icon. You should have those two views visible and see the building blocks etc..

**Reply by msevestre** [reply, marked as answer]
if this is not the case, your layout is somewhat corrupted and you can follow the step
https://github.com/Open-Systems-Pharmacology/MoBi/issues/322#issuecomment-418707361

but make sure you use the latest version of the software (11.2 in your case)
 "C:\Users\ **USERID** \AppData\Roaming\Open Systems Pharmacology\MoBi\11.2\Settings.xml"

**Reply by SamLeDruide** [reply]
Many many thanks, I had to remove the file "settings.xml" but now it works perfectly, I have access to full MoBi as before and I can save my work!

Thanks again for your support. Very helpfull!!

**Comment by StephanSchaller** [comment]
>  Probably sthg that should be removed

I would agree. At least the in-(non-formula)dependent parameters
